src/main/scala/cromwell/engine/backend/BackendCall.scala

```diff
@@ -54,21 +54,37 @@ trait ExecutionHandle {
   def result: ExecutionResult
 }
 
+sealed trait ReturnCode {
+  def code: Int
+
+  def asScriptReturnCode = this match {
+    case script: ScriptReturnCode => Option(code)
+    case _ => None
+  }
+
+  def asBackendReturnCode = this match {
+    case script: BackendReturnCode => Option(code)
+    case _ => None
+  }
```

Comment:
This feels like an odd abstraction to me. Isn't this what an Either is for?

But on the other hand, can't a call have both a ScriptReturnCode AND a BackendReturnCode?